sources/engine/Stride.Engine.Tests/TestUpdateEngine.cs

```diff
@@ -33,7 +34,7 @@ static bool CanPin<T>()
                     return false;
                 }
             }
-            Assert.False(CanPin<AutoLayout>());
+            Assert.True(CanPin<AutoLayout>());
```

Comment:
Oh you pinged me a while ago. I must've completely missed the mention. Sorry about that @Eideren.

I'm pretty sure if it was part of the PR and got accepted, the test passed. And indeed it does on .NET Framework, which is the easiest for me to test without installing old .NET runtimes. It is very possible that it also passes on older .NET Core or .NET 5/6. But if it now doesn't pass and you adjusted it to that circumstance, it'll likely be fine.

It is testing behavior of the runtime and that appears to have changed sometime. It was convenient that this test returned false just to know that in some places nobody forgot to change struct layout from the default of auto, at least where pinning was attempted - I believe this could be the case in the animation infrastructure. I don't think there is any other impact of this change in behavior.

Reply:
Coincidentally I come accross another situation where a test crashes and because of this here, I checked and it happens that it is 2024 when I learn that already on .NET 2.0 an unannotated struct is in fact layout sequential. I must've seen this so many times in ILSpy or wherever... Not knowledge active enough to not surprise me. Somehow I assumed it to be auto, probably when writing the above code. At least when I wrote the reply...